rust/decentraland-godot-lib/src/godot_classes/dcl_avatar.rs

```diff
@@ -0,0 +1,175 @@
+
+    #[var]
+    current_parcel_scene_id: i32,
+
+    #[var]
+    current_parcel_position: Vector2i,
+
+    lerp_state: LerpState,
+    #[base]
+    base: Base<Node3D>,
+}
+
+#[godot_api]
+impl Node3DVirtual for DclAvatar {
+    fn init(base: Base<Node3D>) -> Self {
+        Self {
+            movement_type: AvatarMovementType::ExternalController,
+            current_parcel_scene_id: SceneId::INVALID as i32,
+            current_parcel_position: Vector2i::new(i32::MAX, i32::MAX),
+            lerp_state: Default::default(),
+            base,
+        }
+    }
+}
+
+#[godot_api]
+impl DclAvatar {
+    #[signal]
+    fn change_parcel_position(&self, parcel_position: Vector2) {}
+
+    #[signal]
+    fn change_scene_id(&self, new_scene_id: i32, prev_scene_id: i32) {}
+
+    #[func]
+    pub fn set_target_position(&mut self, new_target: Transform3D) {
+        self.lerp_state.target_distance = self
+            .lerp_state
+            .target_position
+            .distance_to(new_target.origin);
+
+        self.lerp_state.initial_position = self.lerp_state.target_position;
+        self.lerp_state.target_position = new_target.origin;
+        self.lerp_state.factor = 0.0;
+
+        // TODO: check euler order
+        self.base
+            .set_global_rotation(new_target.basis.to_euler(EulerOrder::YXZ));
+        self.base
+            .set_global_position(self.lerp_state.initial_position);
+
+        self.update_parcel_position(self.lerp_state.target_position);
+    }
+
+    #[func]
+    pub fn update_parcel_position(&mut self, position: Vector3) -> bool {
+        let godot_parcel_position = position / 16.0;
+        let parcel_position = Vector2i::new(
+            f32::floor(godot_parcel_position.x) as i32,
+            f32::floor(-godot_parcel_position.z) as i32,
+        );
+
+        if self.current_parcel_position != parcel_position {
+            self.current_parcel_position = parcel_position;
+            self.base.call_deferred(
+                "emit_signal".into(),
+                &[
+                    "change_parcel_position".to_variant(),
+                    parcel_position.to_variant(),
+                ],
+            );
+
+            let scene_runner = DclGlobal::singleton().bind().get_scene_runner();
+            let scene_id: i32 = scene_runner
+                .bind()
+                .get_scene_id_by_parcel_position(parcel_position);
+
+            if self.current_parcel_scene_id != scene_id {
+                let prev_scene_id = self.current_parcel_scene_id;
+                self.current_parcel_scene_id = scene_id;
+                self.base.call_deferred(
+                    "emit_signal".into(),
+                    &[
+                        "change_scene_id".to_variant(),
+                        scene_id.to_variant(),
+                        prev_scene_id.to_variant(),
+                    ],
+                );
+            }
+            return true;
+        } else if self.current_parcel_scene_id == -1 {
```

Comment:
```suggestion
        } else if self.current_parcel_scene_id == SceneId::INVALID {
```

or SceneId::EMPTY if you apply my previous feedback